core/src/main/java/io/questdb/cairo/RebuildColumnBase.java

```diff
@@ -264,37 +265,49 @@ private void reindexOneOrAllColumns(
             long partitionTimestamp,
             long partitionSize
     ) {
+        boolean isIndexed=false;
         tempStringSink.clear();
         partitionDirFormatMethod.format(partitionTimestamp, null, null, tempStringSink);
 
-        if (columnIndex == REBUILD_ALL_COLUMNS) {
-            for (int i = 0, n = metadata.getColumnCount(); i < n; i++) {
-                if (isSupportedColumn(metadata, i)) {
+        for (int i = 0, n = metadata.getColumnCount(); i < n; i++){
+            if(isSupportedColumn(metadata,i)){
+                isIndexed=true;
+                break;
+            }
+        }
+
+        if(isIndexed) {
+            if (columnIndex == REBUILD_ALL_COLUMNS) {
+                for (int i = 0, n = metadata.getColumnCount(); i < n; i++) {
```

Comment:
It's much simpler to introduce `boolean hasIndexedColumns = false;` variable before this loop and assign it to `true` if there is at least one indexed column. Then we could have the following check after the loop:
```java
if (!hasIndexedColumns) {
    throw CairoException.nonCritical().put(unsupportedTableMessage);
}
```

Reply:
I had thought about the same approach but in the scenario where the REINDEX operation is called in a column based manner the flow will not go into this loop and it will display the error messade " The column does not have any index" . The issue had specified to have a consistent error message "Table does not have any index" for both the table based and column based REINDEX operation.

Alternatively the approach suggested by you also solves the issue where success message is displayed when the table wise REINDEX operation is performed. Even though the error message will be different for both table based and column based REINDEX operation ,this approach will solve the issue addressed.

If the consistent message for both the flows is not required I will go ahead and make the changes as suggested by you.

Reply:
Sure will make the changes and push again.